addons/product/report/product_product_templates.xml

```diff
@@ -104,13 +104,18 @@
                         <t t-if="barcode">
                             <!-- `quiet=0` to remove the left and right margins on the barcode -->
                             <div t-out="barcode" style="padding:0" t-options="{'widget': 'barcode', 'quiet': 0, 'symbology': 'auto', 'img_style': barcode_size}"/>
-                            <div class="o_label_name" style="line-height: 130%;height:2.7em;background-color: transparent;">
+                            <div class="o_label_name" style="line-height: 130%;height:2.0em;background-color: transparent;">
                                 <span t-out="barcode"/>
                             </div>
                         </t>
                     </div>
-                    <div class="o_label_name" style="line-height: 130%;height:2.7em;background-color: transparent;">
-                        <span t-field="product.display_name"/>
+                    <div class="o_label_name" style="line-height: 130%;height:2.0em;background-color: transparent;">
+                        <t t-if="product._name == 'product.template'">
+                            <span t-field="product.name"/>
+                        </t>
+                        <t t-else="">
+                            <span t-field="product.display_name"/>
+                        </t>
```

Comment:
```suggestion
                    <div class="o_label_name" style="line-height: 130%;height:2.0em;background-color: transparent;">
                        <span t-if="product._name == 'product.template'" t-field="product.name"/>
                        <span t-else="" t-field="product.display_name"/>
```

Reply:
Done, Thanks for your help 🙏

Reply:
@svs-odoo I think it's better to use `"product.is_product_variant"` instead of `if="product._name == 'product.template'"`